### PR TITLE
buffered car append

### PR DESCRIPTION
**Pull request by turuslan**

### CODE REVIEW

File: core/storage/ipld/cids_ipld.cpp [hunk 1]
```diff
@@ -179,22 +183,59 @@ namespace fc::storage::ipld {
...
+    std::shared_lock lock{car_flush_mutex};
+    const auto it{car_queue.find(row.offset.value())};
+    if (it == car_queue.end()) {
+      return false;
+    }
+    BytesIn item{it->second};
+    BytesIn input;
+    const CbCid *hash;
+    if (codec::uvarint::readBytes(input, item)
+        && codec::cbor::light_reader::readCborBlake(hash, input)) {
```
Comment (wer1st) [inline core/storage/ipld/cids_ipld.cpp:213]: I suggest to rework `readCborBlake()` from out parameter to return based approach.
`const CbCid *hash = readCborBlake(input)`
`if (hash != nullptr) { ... }`

Comment (wer1st) [reply core/storage/ipld/cids_ipld.cpp:213]: @Alexey-N-Chernyshov you have TODO for that in `readCborBlake()`
What do you think?

Comment (Alexey-N-Chernyshov) [reply core/storage/ipld/cids_ipld.cpp:213]: Yes, I prefer not to use out parameters, and here the function like `bool isCborBlakeObject(input)` would be more meaningful to check if input has valid object.
But I think the point of discussion is out of PR scope.

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: core/storage/ipld/cids_ipld.cpp (hunk 1 above)
Comment (wer1st) [reply core/storage/ipld/cids_ipld.cpp:213]: I agree that we could fix it in separate ticket

**Review by wer1st — APPROVED** [review]

**Review by wer1st — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**